collector/partitions.go

```diff
@@ -0,0 +1,264 @@
+// get partitions info of the system
+
+package main
+
+import (
+	"io/ioutil"
+	"os"
+	"path"
+	"strconv"
+	"strings"
+
+	si "github.com/AstroProfundis/sysinfo"
+)
+
+// BlockDev is similar to blkDev_cxt in lsblk (from util-linux)
+// contains metadata of a block device
+type BlockDev struct {
+	Name      string     `json:"name,omitempty"`
+	Partition bool       `json:"partition,omitempty"`
+	Mount     MountInfo  `json:"mount,omitempty"`
+	UUID      string     `json:"uuid,omitempty"`
+	Size      uint64     `json:"size,omitempty"`
+	SubDev    []BlockDev `json:"subdev,omitempty"`
+	Holder    []string   `json:"holder_of,omitempty"`
+	Slave     []string   `json:"slave_of,omitempty"`
+}
+
+// MountInfo is the metadata of a mounted device
+type MountInfo struct {
+	MountPoint string `json:"mount_point,omitempty"`
+	FSType     string `json:"filesystem,omitempty"`
+	Options    string `json:"mount_options,omitempty"`
+}
+
+const sysClassBlock = "/sys/block"
+
+func GetPartitionStats() []BlockDev {
+	partStats := make([]BlockDev, 0)
+	if dirSysBlk, err := os.Lstat(sysClassBlock); err == nil &&
+		dirSysBlk.IsDir() {
+		fi, err := os.Open(sysClassBlock)
+		if err != nil {
+			printErr(err)
+		}
+		blockDevs, err := fi.Readdir(0)
+		if err != nil {
+			printErr(err)
+		}
+		for _, blk := range blockDevs {
+			var blkDev BlockDev
+			if blkDev.getBlockDevice(blk, nil) {
+				partStats = append(partStats, blkDev)
+			}
+		}
+		matchUUIDs(partStats, checkUUIDs())
+		matchMounts(partStats, checkMounts())
```

Comment:
you should not change partStats in matchUUIDs. By the name, It is just match not changing.
even you use name changing, you shoule change it in function `GetPartitionStats`

Reply:
It is about communicational cohesion . do not change it out of this function